Move to core 22 (5.0-edge)

mihalicyn opened the pull request:
- move to core22
- fix ZFS 0.6/0.7 build issues on core22

stgraber committed:
snapcraft: Fix meson builds

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
(cherry picked from commit a2c01af1d89d0f257b14e431aeee024f713259ee)
Signed-off-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>

stgraber committed:
snapcraft: Move to core22

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
(cherry picked from commit e85fe8bdf1d3dcefb90d3bfe3642ee49c2430fee)
Signed-off-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>

stgraber committed:
nvidia-container: Tweak for 22.04

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
(cherry picked from commit c20fbd98c794e0047eea653b33f695b86cd855c7)
Signed-off-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>

stgraber committed:
nvidia-container: Fix issue with Ubuntu 20.04

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
(cherry picked from commit e98676c5301bc997fbf0dc5a274798fc41a58abb)
Signed-off-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>

mihalicyn commented:
ZFS 0.6:
https://github.com/mihalicyn/zfs/commits/zfs-0.6-core22-compilation-fix
 
ZFS 0.7:
https://github.com/mihalicyn/zfs/commits/zfs-0.7-core22-compilation-fix

I'm on the way to create `.patch` files and properly include them in the `lxd-pkg-snap`. But just for demonstration and testing purposes publishing this PR as it is.

mihalicyn committed:
zfs: fix build for zfs 0.6 and 0.7 on core22

Signed-off-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>

simondeziel commented on snapcraft.yaml:
```diff
@@ -1054,6 +1049,12 @@ parts:
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-6/0002-config-better-libtirpc-detection.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-6/0003-explicitly-define-____cacheline_aligned.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-6/0004-Fix-static-data-to-link-with-fno-common.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-6/0005-A-simplified-backport-of-2faf05612f9-libshare-smb-cl.patch"
```
`s/SNAPCRAFT_PROJECT_DIR/CRAFT_PROJECT_DIR/` here too, please.

simondeziel replied:
Ah, I just realize you used the old variable name to be consistent with what was done in that part, please ignore my comment then. We'll change them all in a later PR.

simondeziel commented on snapcraft.yaml:
```diff
@@ -1088,6 +1089,10 @@ parts:
 
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-7/0001-explicitly-define-____cacheline_aligned.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-7/0002-Fix-static-data-to-link-with-fno-common.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-7/0003-A-simplified-backport-of-2faf05612f9-libshare-smb-cl.patch"
```
`s/SNAPCRAFT_PROJECT_DIR/CRAFT_PROJECT_DIR/` here too, please.

simondeziel replied:
Same here, please ignore.

simondeziel reviewed (dismissed):
LGTM, with the small inconsistency `s/SNAPCRAFT_PROJECT_DIR/CRAFT_PROJECT_DIR/` fixed.

That is a very welcome change, thank you!

tomponline commented:
LGTM, but wont merge until after LXD 5.0.3 is released.

tomponline commented:
This can be rebased and we can then look to merge it soon. Ta

simondeziel commented:
This would also be a good time to drop the `build-attributes` that no longer make sense if the base is `core22`, see https://snapcraft.io/docs/release-notes-snapcraft-6-1#heading--core22

mihalicyn commented:
> This can be rebased and we can then look to merge it soon. Ta

Have rebased.

mihalicyn committed:
snapcraft: remove build-attributes for core22-step-dependencies

It makes no sense anymore as we step on core22.

Suggested-by: Simon Deziel <simon.deziel@canonical.com>
Signed-off-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>

mihalicyn commented:
> This would also be a good time to drop the `build-attributes` that no longer make sense if the base is `core22`, see https://snapcraft.io/docs/release-notes-snapcraft-6-1#heading--core22

Have done.
